### PR TITLE
Limit number of pgf stewards

### PR DESCRIPTION
**Pull request by Fraccaman**
## Describe your changes
Introduced a new genesis parameter to limit the number of pgf stewards.

## Indicate on which release or other PRs this topic is based on
0.39.0

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Review by brentstone — DISMISSED** [review]

File: .changelog/unreleased/imrpovements/3442-limit-pgf-stewards.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Enforce an upper limit on the number of PGF stewards allowed to exist at a
```
Comment (grarco) [inline .changelog/unreleased/imrpovements/3442-limit-pgf-stewards.md:1]: Small typo in the name of this file `imrpovements`

File: crates/node/src/shell/governance.rs [hunk 2]
```diff
@@ -461,18 +467,41 @@ fn execute_pgf_steward_proposal<S>(
...
+    }
+
+    // Then add new addresses
+    let mut steward_count = pgf_storage::stewards_handle().len(storage)?;
+    for address in stewards.iter().filter_map(|action| match action {
+        AddRemove::Add(address) => Some(address),
+        AddRemove::Remove(_) => None,
+    }) {
+        #[allow(clippy::arithmetic_side_effects)]
+        if steward_count + 1 > maximum_number_of_pgf_steward {
```
Comment (grarco) [inline crates/node/src/shell/governance.rs:493]: Should this be a checked operation?

File: crates/node/src/shell/governance.rs [hunk 3]
```diff
@@ -461,18 +467,41 @@ fn execute_pgf_steward_proposal<S>(
...
+        }
+        pgf_storage::stewards_handle().insert(
+            storage,
+            address.to_owned(),
+            StewardDetail::base(address.to_owned()),
+        )?;
+
+        #[allow(clippy::arithmetic_side_effects)]
+        {
+            steward_count += 1;
```
Comment (grarco) [inline crates/node/src/shell/governance.rs:504]: Same here

File: crates/node/src/shell/governance.rs (hunk 2 above)
Comment (Fraccaman) [reply crates/node/src/shell/governance.rs:493]: the limit will be a number < 100 and always > 0 so making the operation checked here is too much I think

File: crates/node/src/shell/governance.rs (hunk 3 above)
Comment (Fraccaman) [reply crates/node/src/shell/governance.rs:504]: same as above

File: .changelog/unreleased/imrpovements/3442-limit-pgf-stewards.md (hunk 1 above)
Comment (brentstone) [reply .changelog/unreleased/imrpovements/3442-limit-pgf-stewards.md]: fixed thx

**Pull request merged**